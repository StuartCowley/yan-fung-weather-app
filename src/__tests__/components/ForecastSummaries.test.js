import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
  const validProps = [
    {
      date: 11100011,
      description: "Stub description 1",
      icon: "stubIcon1",
      temperature: {
        min: 12,
        max: 22,
      },
    },
    {
      date: 22200011,
      description: "Stub description 2",
      icon: "stubIcon2",
      temperature: {
        min: 15,
        max: 27,
      },
    },
  ]; 

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct number of ForecastSummary instances", () => {
    const { getAllByTestId } = render(
        <ForecastSummaries forecasts={validProps} />
    );

    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
  })
})