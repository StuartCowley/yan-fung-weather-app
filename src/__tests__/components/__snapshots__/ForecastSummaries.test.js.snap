// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu, 1 Jan
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 1
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        <img
          alt="thermometer logo"
          height="25px"
          src="thermometer.png"
          width="25px"
        />
        22°C
      </div>
      <button
        class="button"
        type="button"
      >
        MORE
      </button>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        Thu, 1 Jan
      </div>
      <div
        class="forecast-summary__description"
      >
        Stub description 2
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-snow"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        <img
          alt="thermometer logo"
          height="25px"
          src="thermometer.png"
          width="25px"
        />
        24°C
      </div>
      <button
        class="button"
        type="button"
      >
        MORE
      </button>
    </div>
  </div>
</DocumentFragment>
`;
